plugins/worker/lib/jobs.js

```diff
@@ -136,12 +136,13 @@ async function pushToRabbitMq(message, queue, messageId) {
  * Push message to Kafka topic
  * @param {Object} message  Job and build config metadata
  * @param {String} topic          Topic name
+ * @param {String} messageId The message id
  */
-async function pushToKafka(message, topic) {
-    const conn = await AWSProducer.connect();
+async function pushToKafka(message, topic, messageId) {
+    const producer = await AWSProducer.connect();
 
-    if (conn) {
-        await AWSProducer.sendMessage(message, topic);
+    if (producer) {
```

Comment:
`pushToKafka` function is invoked only when kafka is enabled. In that case `AWSProducer.connect()` must return producer object (non null).
Why would we need this check?
Is it possible that `AWSProducer.connect()` would resolve to `null` even when kafka config is enabled. If yes, isn't that an undesired behavior?

Reply:
it can be null, if some setting are wrong, but queue service should not crash

Reply:
Should we log error in this case to detect and act on this?

Reply:
it will be logged as part of the package